src/gitlab.js

```diff
@@ -78,6 +82,88 @@ class GitLab {
     });
   }
 
+  loadBranchs(cur_proj_name) {
+    var projId = encodeURIComponent(cur_proj_name);
+    m.request({
+      url: `${this.api_path}/projects/${projId}`,
+      method: "GET",
+      headers: {
+        "PRIVATE-TOKEN": this.private_token
+      }
+    }).then((proj) => {
+      this.cur_proj_name = proj.name_with_namespace;
+      this.cur_proj_id = proj.id;
+      this.loadTriggers();
+      this.branchs = null;
+      return this.loadBranchsBase(1, []);
+    }).catch((e) => {
+      alert(e);
+      return Promise.reject();
+    });
+  }
+  loadTriggers() {
+    m.request({
+      url: `${this.api_path}/projects/${this.cur_proj_id}/triggers`,
+      method: "GET",
+      headers: {
+        "PRIVATE-TOKEN": this.private_token
+      }
+    }).then((trig) => {
+      this.triggers = trig;
+      return Promise.resolve(trig);
+    }).catch((e) => {
+      alert(e);
+      return Promise.reject();
+    });
+  }
+  
+
+  loadBranchsBase(page, all_branchs) {
+    const data = {
+      page: page,
+      per_page: this.per_page,
+      order_by: "name",
+      sort: "asc"
+    };
+
+    if (this.apiVersion >= 4) {
+      // Until v3, GET /projects returns that user is member.
+      // But since v4, GET /projects returns all projects visible to current user, even if the user is not a member.
+      // To get projects the user is a member of, use GET /projects?membership=true
+      data.membership = true;
+    }
+
+    // List repository branches 
+    // GET /projects
+    // https://github.com/gitlabhq/gitlabhq/blob/master/doc/api/projects.md#list-projects
+    // NOTE: order_by and sort are supported by v7.7.0+. If no options, order_by created_at DESC
+    return m.request({
+      url: `${this.api_path}/projects/${this.cur_proj_id}/repository/branches`,
+      method: "GET",
+      data: data,
```

Comment:
`GET /projects/:id/repository/branches` receives no query string

see. https://docs.gitlab.com/ce/api/branches.html#list-repository-branches

Reply:
I didn't get this one.   

I can receive a json array by use this API, and use *branch.name* in a dropdown menu for each branch.

Reply:
I see